plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1985,8 +1985,8 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         Table table = metastore.getTable(handle.getSchemaName(), handle.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(handle.getSchemaTableName()));
         if (table.getTableType().equals(MANAGED_TABLE.name()) &&
-                !(metastoreType.equalsIgnoreCase("glue") || metastoreType.equalsIgnoreCase("file"))) {
-            throw new TrinoException(NOT_SUPPORTED, format("Renaming managed tables is not supported for %s metastore", metastoreType));
+                !allowTableRename) {
+            throw new TrinoException(NOT_SUPPORTED, "Renaming managed tables is not allowed");
```

Comment:
in https://github.com/trinodb/trino/pull/14059 i went with "with current metastore configuration" to indicate this isn't a permanent Delta's property, but it's configuration dependent. 
Not sure if it matters though